File: 3.x.x/crates-utopia/domains/src/registrar/update_details.rs
/// Domain update payload (PHP `Utopia\Domains\Registrar\UpdateDetails`).
#[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
pub struct UpdateDetails {
    pub auto_renew: Option<bool>,
}

impl UpdateDetails {
    /// PHP constructor (`$autoRenew = null`).
    pub fn new(auto_renew: Option<bool>) -> Self {
        Self { auto_renew }
    }
}

File: packages/domains/src/Domains/Registrar/UpdateDetails.php
<?php

declare(strict_types=1);

namespace Utopia\Domains\Registrar;

final class UpdateDetails
{
    /**
     * @param bool|null $autoRenew Enable or disable automatic renewal
     */
    public function __construct(
        public ?bool $autoRenew = null,
    ) {}
}
